core/logger.go

```diff
@@ -0,0 +1,37 @@
+package core
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// Logger provides a simple definition of Logging methods that may be used within this project.
+// Logger implementations can be set by getting the current logger referent with GetLogger and to replace it with one's
+// own implementation.
+type Logger interface {
```

Comment:
The ideal interface is small and composeable. This is a rather large interface which is less composable. For instance I take that `Print()` is supposed to always print something to attached stream (like Stdout on a TTY) because it is _unleveled_. Print does not contain a level of semantic verbosity like a log level DEBUG does.

In turn, why would a K8s component need to implement this method, when it is useless?

I am not going to lie, I took a deep dive through this wild discussion [about a golang log interface proposal](https://groups.google.com/g/golang-dev/c/F3l9Iz1JX4g/m/szAb07lgFAAJ) and [Chris Hines's findings](https://go-talks.appspot.com/github.com/ChrisHines/talks/structured-logging/structured-logging.slide)

Following his idea, maybe an alternative Logger interface WOULD look like a single method interface consisting solely of 
```golang
  Log(args...) error
```
while the rest (namely the mapping of levels, selections of streams, etc) stays implementation specific?

Anyhow, while Chris Hines's [gokit library](https://github.com/go-kit/log/blob/main/level/example_test.go) tries to be generic it might not meet our standards of fitting half a dozen of totally different frameworks under the same hood.

I still like the idea of getting a log level first and then provide the log arguments, like [in this example](https://github.com/go-kit/log/blob/main/level/example_test.go#L14). This may syntax could allow us to wrap a everything-agnostic logger into a use-case-specific level oriented construct.

I still work on our use-cases of CLI/containers/library logs that should work hand in hand.